### ISSUE TITLE
[Change request]: Change academic-sessions to find start- and enddate of an academic year

### ISSUE THREAD

**Issue by hermanvand** [open-education-api/specification#161]
### Organization

SURF

### Project

eduXchange.nl

### Contact Details

herman.vandompseler@surf.nl

### Short description

The academic year does not exactly start on the 1st of september and varies per institution. In case of the WUR for example, in 2021 it started on 30 8 2021. UU started that year on 6 9 2021. To get this startdate (and endate) there is a need to query the academic-sessions of institutions and get academic years in return.

### Version

v4

### Usecase

In eduXchange we use academic year and startdate to find courses. The start of our academic year is fixed at 1/9. But in 2021 this does not work out very well for WUR where courses start at 30/8. Therefor we need to fetch the start of the academic year on an institution base. The academic sessions seem to be the logical place to do so.

### Proposed solution

**Current situation**

The current academic-sessions structure is a tree structure of which the root object is an academic-year, that can have many children. It could look like:

- year (is a academic-session)
  - academic-session
  - parent (is a academic-session)
    - academic-session

the 'parent' and 'year' are objects in an academic-session. But they are not required.

In the current OOAPI spec it is described that an academic-session without a  parent attribute is the academic year. The description of 'parent' is:  

_"Filter by parent (academicSessionId), may be empty to retrieve all years (which have no parent)"_

btw: the year object could be used to find an academic year for a particular academic-session in the tree, this has nothing to do with this use case.

**Problem**

The current desciption assumes that the parent object is always filled correctly, although it is not required, and a filter by parent="" will actually work.

Our experience is that this is not always the case. A request with parent="" (/academic-sessions?parent="") could also be interpreted as a request without a parent attribute (/academic-sessions), but that has different results.

**Example**

This is an example of an institution, WUR, describing an academic year

```
{
"academicSessionId": "483ec6ae-66a3-44c5-9dd8-cbda210238e3",
"name": "Academic Year",
"startDate": "2021-08-30",
"endDate": "2022-08-31",
"ext": {}
}
```

**Solution**

There are two solutions to get the academic year out of a set of academic-sessions properly. These are:

1. We make the parent attribute required and name the 'root' parent explicitly when creating an Academic Year. So an academic year can be found with '?parent=root'
2. we introduce a new attribute "isYear=true (default is false)" and make it possible to search with this filter: academic-sessions?isYear=true

Solution 1 resembles the solution the OOAPI also uses for the root organization. The similar query is /organizations?type=root. Solution 2 is straight to the point and could work equally well.

We considered other solutions, but they seem more difficult to realize and introduce new problems:

3. We decide on a particular 'name', for example 'Academic Year' and find years by that name.
4. we introduce a search parameter to find an academic-session without a year object. For example: academic-sessions?year=false. In this case we assume all other academic-sessions have a year specified. 
5. we could introduce a search query to find a specfic academic year based on a given date, for example: academic-sessions?getYearWithDate=20220201. This requires more logic in the api

### Requests and responses

/academic-sessions

**Comment by mpdejong** [comment]
I have looked into the definition of AcademicSession in IMS Global Edu-API (release candidate v1).

In the Edu-API the issue of root or top level AcademiSession has been solved by introducing a sessionType. 
On the issue with the search query the Edu-API is using a twoway relationship possibility. So not only a parent, but also children. 

AcademicSession has the following optinal attributes that might be helpful in this use case
- sessionType -> AcademicSessionTypeEnum Enumeration  [Optional] (This is an extendable enumation, so proprietary terms can be used)
- parent -> AcademicSessionUUIDRef [Optional]
- children -> AcademicSessionUUIDRef [Optional]
